GeosCore/hco_utilities_gc_mod.F90

```diff
@@ -1897,85 +1896,40 @@ SUBROUTINE Get_GC_Restart( Input_Opt, State_Chm, State_Grid, &
-      IF ( RC /= GC_SUCCESS ) THEN
-         Msg = 'Error returned from Convert_Spc_Units, call #1!'
-         CALL GC_Error( Msg, RC, Loc )
-         RETURN
-      ENDIF
-
-      ! Start HEMCO timer again
-      IF ( Input_Opt%useTimers ) THEN
-         CALL Timer_Start( "HEMCO", RC )
-      ENDIF
-
-      ! Print values
-      DO N = 1, State_Chm%nSpecies
-         SpcInfo => State_Chm%SpcData(N)%Info
-         WRITE(6,150) N, TRIM( SpcInfo%Name ),         &
-                         MINVAL( Spc(N)%Conc(:,:,:) ), &
-                         MAXVAL( Spc(N)%Conc(:,:,:) )
-150      FORMAT( 'Species ', i3, ', ', a9,             &
-                 ': Min = ', es15.9, ', Max = ', es15.9 )
-         SpcInfo => NULL()
-      ENDDO
-
-      ! Halt HEMCO timer (so that unit conv can be timed separately)
-      IF ( Input_Opt%useTimers ) THEN
-         CALL Timer_End( "HEMCO", RC )
-      ENDIF
-
-      ! Convert units back
-      CALL Convert_Spc_Units(                                                &
-           Input_Opt  = Input_Opt,                                           &
-           State_Chm  = State_Chm,                                           &
-           State_Grid = State_Grid,                                          &
-           State_Met  = State_Met,                                           &
-           new_units  = previous_units,                                      &
-           RC         = RC                                                  )
-
-      ! Trap error
-      IF ( RC /= GC_SUCCESS ) THEN
-         Msg = 'Error returned from Convert_Spc_Units, call #2!'
-         CALL GC_Error( Msg, RC, Loc )
-         RETURN
-      ENDIF
-
-      ! Start HEMCO timer again
-      IF ( Input_Opt%useTimers ) THEN
-         CALL Timer_Start( "HEMCO", RC )
+   !=========================================================================
+   ! Get delta pressure per grid box stored in restart file to allow mass
+   ! conservation across consecutive runs
+   !========================================================================
+   v_name_in_hemco = 'DELPDRY'
+   v_name_in_file  = 'Met_DELPDRY'
+
+   ! Get variable from HEMCO and store in local array
+   CALL HCO_GC_GetPtr( Input_Opt, State_Grid, TRIM( v_name_in_hemco ),  &
+        Ptr3D,     RC,         FOUND=FOUND )
+
+   IF ( FOUND ) THEN
+      State_Met%DELP_DRY = Ptr3D
+      IF ( Input_Opt%amIRoot ) THEN
+         WRITE( 6, 510 ) ADJUSTL( v_name_in_file ),           &
+              MINVAL(  State_Met%DELP_DRY ),                  &
+              MAXVAL(  State_Met%DELP_DRY ),                  &
+              SUM(     State_Met%DELP_DRY )
       ENDIF
+   ELSE
+      State_Met%DELP_DRY = 0.0_fp
+      IF ( Input_Opt%amIRoot ) WRITE( 6, 520 ) ADJUSTL( v_name_in_file )
    ENDIF
 
+   ! Nullify pointer
+   Ptr3D => NULL()
+
    !=========================================================================
    ! Get variables for KPP mechanisms (right now just fullchem and Hg)
```

Comment:
The comment should read `(right now fullchem, Hg, and carbon)'

Reply:
This is unrelated to the PR changes. Could that update go into docs/dev?

Reply:
Actually, looking at this code, are you sure it is supposed to include carbon? The section is to retrieve `KPP_HVALUE` from the restart file. If it should include carbon then there is a bug that would need to be fixed.